src/sknnr/_weighted_trees.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+from typing import Callable, Literal
+
+import numpy as np
+from numpy.typing import ArrayLike, NDArray
+
+from ._base import TransformedKNeighborsRegressor, YFitMixin
+from .transformers import RFNodeTransformer
+
+TREE_NODE_TRANSFORMER = RFNodeTransformer
+
+
+class WeightedHammingDistanceMetric:
+    """
+    Distance metric to calculate Hamming distance based on weights.
+    Setting of weights is only supported after initialization.
+    """
+
+    def __init__(self):
+        self.w = None
+
+    def set_weights(self, w: ArrayLike[float]):
+        self.w = w
+
+    def __call__(self, u: NDArray, v: NDArray) -> float:
+        if self.w is None:
+            self.w = np.ones_like(u)
+        return np.sum((u != v) * self.w) / np.sum(self.w)
```

Comment:
Would there be a disadvantage to using `scipy.spatial.distance.hamming` here? If not, that seems slightly simpler and presumably a little more consistent with the Hamming distance you'd see elsewhere in `sklearn` (I think you mentioned that this returned *slightly* different results).